src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -909,6 +909,7 @@ mongoc_collection_estimated_document_count (
          /* Collection does not exist. From spec: return 0 but no err:
           * https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#estimateddocumentcount
           */
+         bson_reinit (reply);
```

Comment:
the reply may be null but `bson_reinit` asserts that it isn't before attempting to reinitialize it. The simplest solution would be to make sure you have a reply before call `bson_reinit`.

Reply:
Great point! Added in a null check.